rust/common/types/src/team.rs

```diff
@@ -0,0 +1,63 @@
+use serde::{Deserialize, Serialize};
```

Comment:
in feature flags land, our team module implements methods to fetch teams from either PG or Redis – is the idea here that we don't want the concept of teams to exist in Redis for error tracking?  IMO if we already have the info stored in Redis for a given team, it seems faster to just pull the data from there instead of use Postgres for everything. Why not do that here?

Like I'm not necessarily encouraging that we don't use this team cache as a persistent storage of team data that runs alongside the service and saves us hops to external dependencies as it starts to fill up, I'm saying that why not give it the ability to pull from Redis whenever we're inserting new tokens into it?

Reply:
There's no reason not to extend the `impl` below to be able to hit redis as well, but it struck me as a premature optimisation, if we're also doing in-memory caching for teams (and per-team data is small, and PG reads are ~fast for small tables). 

I'd keep hitting PG and hitting redis as separate operations, just to make it more explicit what you're doing (I wouldn't make them a single function)

Reply:
this optimisation is necessary for flags, unless not using pgbouncer with rust ensures we have sufficient connections for all parallel requests to pg that also don't timeout. [This is a common failure mode where we don't want flags to go down]. This wasn't possible for flag scale.

Reply:
oh although if there's a persistent in memory cache, which means we don't need to go to pg or redis both, that can indeed work, assuming flag updates / team updates are kept in sync somehow. This wasn't easily possible iirc so django never tried to maintain this

Reply:
The in memory cache just invalidates after some time - the only thing that would invalidate it is token rotation, and given this is ingest anyway, some lag is expected.

Reply:
Totally reasonable for ingestion 👍